src/libsyntax/parse/parser.rs

```diff
@@ -4919,7 +4920,8 @@ impl<'a> Parser<'a> {
 
     /// Parse a method or a macro invocation in a trait impl.
     fn parse_impl_method(&mut self, vis: &Visibility, at_end: &mut bool)
-                         -> PResult<'a, (Ident, Vec<ast::Attribute>, ast::ImplItemKind)> {
+                         -> PResult<'a, (Ident, Vec<ast::Attribute>, ast::Generics,
+                             ast::ImplItemKind)> {
```

Comment:
I had to make this return type even larger since Generics is outside MethodSig now. There was a lot of code relying on that structure.

Reply:
yep that is expected